models/ed/R/model2netcdf.ED2.R

```diff
@@ -937,7 +937,9 @@ read_E_files <- function(yr, yfiles, efiles, outdir, start_date, end_date, pft_n
   
   npft <- length(pft_names)
   data(pftmapping, package = "PEcAn.ED2")
-  pfts <- sapply(pft_names, function(x) pftmapping$ED[pftmapping$PEcAn == x]) 
+  pfts <- sapply(pft_names, function(x) ifelse(x %in% settings$pfts$pft$name, 
+                                               as.numeric(settings$pfts$pft$ed2_pft_number[settings$pfts$pft$name == x]),
+                                               pftmapping$ED[pftmapping$PEcAn == x])) 
```

Comment:
This has two issues:

- If the PFT `name` is provided but the `ed2_pft_number` is not, it will return `NULL` instead of grabbing the number from the `pftmapping`
- Because of the deep hierarchy of the PEcAn XML PFT section (roughly, something like `settings <- list(pfts = list(pft = list(name = "tree"), pft = list(name = "shrub")))`, it won't work for multiple PFTs.

I might suggest rewriting this block to something like this:

```r
  pfts <- numeric(npft)
  names(pfts) <- pft_names

  # Extract the PFT names and numbers for all PFTs.
  xml_pft_names <- lapply(settings$pfts, "[[", "name")
  for (pft in pft_names) {
    which_pft <- which(xml_pft_names == pft)
    xml_pft <- settings$pfts[[which_pft]]
    # If the PFT number is provided in the XML, use that.
    if ("ed2_pft_number" %in% names(xml_pft)) {
      pft_number <- as.numeric(xml_pft$ed2_pft_number)
      if (!is.finite(pft_number)) {
        PEcAn.logger::logger.severe(
          "ED2 PFT number present but not parseable as number. Value was `",
          xml_pft$ed2_pft_number "`".
        )
      }
    } else {
      # Otherwise, fall back on the PFT number in the `pftmapping`
      pft_number <- pftmapping$ED[pftmapping$PEcAn == x]
    }
    pfts[pft] <- pft_number
  }
```

Reply:
Also, a third issue (caught by our `check` GH Action!): `settings` is not currently passed to this function. A hacky workaround would be to add it as an argument with a default value of `NULL` and modify the calling code to also pass the `settings` argument.

Reply:
instead of sending in the entire settings, it seems that it would be cleaner (and inline w/ prev. discussions) to only send the element of the settings file that is needed. e.g. in this case you could replace two arguments to the function (the pft name and number) with a list object that is either (`pfts = list(pft = list( name = 'SetariaWT', ed2_pft_number = "1"))` or `pfts = settings$pfts`).

Similarly, the model2netcdf.ED function you could have arguments models and pfts, e.g. `, models = settings$models, pfts = settings$pfts`). Not necessary for this PR, but a way to reduce the requirement to have a full settings object (or a list named settings when you just need one or two components) for a particular function - easier testing, easier to understand, and a step closer to package independence.

Reply:
@ashiklom could you please re-review this PR - does it address your requested changes?